### PR TITLE
refactor: update to use Protobom Ent backend

### PR DESCRIPTION
**Pull request by jhoward-lm**
# Update to use Protobom Ent backend

## Description

Update to use the protobom Ent storage backend for interacting with bomctl database.

## Type of change

- [X] Refactor

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by jhoward-lm** [comment]
Going to break this into follow-on issues and separate PRs to keep these changes small.

File: .github/workflows/go-test.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - synchronize
+
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (jhoward-lm) [reply .github/workflows/go-test.yml:37]: Probably! I’ll try adding that in when I open the separate PR for adding the go test workflow

[the remark this answers is not in the document]
Comment (jhoward-lm) [reply .github/workflows/go-test.yml]: For context, here's what I found:

- ARM64 support is still in private beta ([link](https://github.blog/changelog/2023-10-30-accelerate-your-ci-cd-with-arm-based-hosted-runners-in-github-actions/))
- Only a Linux runner is allowed when specifying job container images ([link](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idcontainer))

**Comment by jhoward-lm** [comment]
Putting back into draft until @puerco is able to merge https://github.com/protobom/storage/pull/12 and cut a release so that this repo's `go.mod` can be updated

**Review by ashearin — APPROVED** [review]

**Pull request merged**